src/main/java/com/mappy/fpm/batches/tomtom/dbf/timedomains/TimeDomainsParser.java

```diff
@@ -66,6 +66,26 @@ private String parse(TimeDomains timeDomains) {
         throw new IllegalArgumentException("Unable to parse '" + domain + "'");
     }
 
+    private String padLeftZeros(String inputString, int length) {
+        return String.format("%1$" + length + "s", inputString).replace(' ', '0');
+    }
+
+    public String[] parseDateInterval(TimeDomains timeDomain) {
```

Comment:
I'm really not familiar with java or its standard library, but I'm a bit surprised to see a "manual" date parsing, instead of using a standard parser.

According to the test below, the expected format seems to be `[(y2020M1d15)(y2022M12d5)]`.

Isn't it possible to parse `y2020M1d15` using standard parser ?

Also, as the regex is hard to read, it might help the reader to add a parsed sample in a comment.